### PR TITLE
Update exercises 2 and 3

### PR DESCRIPTION
**Pull request by maryx**
Updates exercises 2 and 3 to be in Dart 2 and updates UX.

Sorry it is so messy looking, and I'm going to have to port the exact same changes all the way up to exercise 5 so there will be a lot of repetitive code.
Currently looking for feedback of 2 and 3 before I move onto 4 and 5; if there are changes I'll make them for the shared code in both exercises.

### CODE REVIEW

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 1]
```diff
@@ -2,79 +2,93 @@
...
+        assert(color != null),
+        super(key: key);
 
-  /// Builds a custom widget that shows unit [Category] information.
+  /// Builds a custom widget that shows [Category] information.
   ///
   /// This information includes the icon, name, and color for the [Category].
   @override
+  // This `context` parameter describes the location of this widget in the
+  // widget tree. It can be used for grabbing Theme data from the nearest
```
Comment (yjbanov) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: nit: grabbing is impolite. Here and below, I'd say "look up" or "obtain" :)

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 2]
```diff
@@ -2,79 +2,93 @@
...
-                  color: color,
+    return Material(
+      color: Colors.transparent,
+      child: Container(
+        height: _rowHeight,
+        child: InkWell(
+          borderRadius: _borderRadius,
+          highlightColor: color,
+          splashColor: color,
+          // We can use either the () => function or the () { function(); }
```
Comment (yjbanov) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: `() => function` would _return_ the function, but `() { function(); }` would _call_ the function. You probably meant `() => function()`.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 3]
```diff
@@ -2,79 +2,93 @@
...
+            print('I was tapped!');
+          },
+          child: Padding(
+            padding: EdgeInsets.all(8.0),
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              // There are two ways to denote a list: `[]` and `List()`.
+              // Prefer to use the literal syntax, i.e. `[]`, instead of `List()`.
+              // You can add the type argument if you'd like, i.e. <Widget>[].
+              // See https://www.dartlang.org/guides/language/effective-dart/usage#do-use-collection-literals-when-possible
```
Comment (yjbanov) [inline course/02_category_widget/solution_02_category_widget/lib/category.dart]: It's really cool that you are providing links to Effective Dart!

**Comment by yjbanov** [comment]
lgtm

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 4]
```diff
@@ -2,79 +2,93 @@
...
+///
+/// The widget is composed on an [Icon] and [Text]. Tapping on the widget shows
+/// a colored [InkWell] animation.
 class Category extends StatelessWidget {
   final String name;
   final ColorSwatch color;
   final IconData iconLocation;
 
-  /// Constructor
+  /// Stores information about a [Category].
```
[the remark this answers is not in the document]
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Done.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart [hunk 5]
```diff
@@ -2,79 +2,93 @@
...
+  ///
+  /// A [Category] saves the name of the Category (e.g. Length), its color for
```
[the remark this answers is not in the document]
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Done.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart (hunk 1 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Ah, makes sense. Done.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart (hunk 2 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Done.

File: course/02_category_widget/solution_02_category_widget/lib/category.dart (hunk 3 above)
Comment (maryx) [reply course/02_category_widget/solution_02_category_widget/lib/category.dart]: Thanks! I think it'd be helpful to them.

**Pull request merged**